scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -3,117 +3,534 @@
+                                    count == 3
 
+    -- Crimson Key: Trade Seedspall's Lux, Luna, Astrum
     if
-        xi.settings.main.ENABLE_ACP == 0 and
-        xi.settings.main.ENABLE_AMK == 0 and
-        xi.settings.main.ENABLE_ASA == 0
+        hasCrimsonItems and
+        finishedACP
     then
-        player:showText(npc, ID.text.GET_LOST)
-
-    -- Crimson Key: Trade Seedspall's Lux, Luna, Astrum
-    else
+        local ki = xi.ki.CRIMSON_KEY
         if
-            xi.settings.main.ENABLE_ACP == 1 and
-            sLux and
-            sLuna and
-            sAstrum and
-            count == 3 and
-            acpMission >= xi.mission.id.acp.GATHERER_OF_LIGHT_I and
-            not hasCrimsonKey and
-            os.time() > lastCrimson
-        then -- and timer stuff here) then
+            not player:hasKeyItem(ki) and
+            now >= player:getCharVar('LastCrimsonKey')
+        then
             player:tradeComplete()
-            player:addKeyItem(xi.ki.CRIMSON_KEY)
-            player:setCharVar('LastCrimsonKey', getMidnight())
+            player:addKeyItem(ki)
+            player:setCharVar('LastCrimsonKey', nextMidnight)
             player:messageSpecial(ID.text.DRYEYES_2)
-            player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.CRIMSON_KEY)
-
-        elseif
-            sLux and
-            sLuna and
-            sAstrum and
-            count == 3 and
-            (os.time() <= lastCrimson or hasCrimsonKey)
+            player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+        else
+            player:messageSpecial(ID.text.DRYEYES_3, ki)
+        end
+    -- White Coral Key: trade 3 drops - orcish plate armor, quadav backscale, yagudo caulk
+    elseif
+        hasWhiteCoralItems and
+        finishedAMK
+    then
+        local ki = xi.ki.WHITE_CORAL_KEY
+        if
+            not player:hasKeyItem(ki) and
+            now >= player:getCharVar('LastWhiteCoralKey')
         then
-            player:messageSpecial(ID.text.DRYEYES_3, xi.ki.CRIMSON_KEY)
-        -- White Coral Key:
-        -- elseif xi.settings.main.ENABLE_AMK == 1 and
-            -- haven't even started AMK related trades yet.
+            player:tradeComplete()
+            player:addKeyItem(ki)
+            player:setCharVar('LastWhiteCoralKey', nextMidnight)
+            player:messageSpecial(ID.text.DRYEYES_2)
+            player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+        else
+            player:messageSpecial(ID.text.DRYEYES_3, ki)
         end
+    -- Moogle Key: trade proper enfeebling kit
+    elseif
+        hasEnfeebKit and
+        finishedASA
+    then
+        local ki = xi.ki.MOOGLE_KEY
+        if
+            not player:hasKeyItem(ki) and
+            now >= player:getCharVar('LastMoogleKey')
+        then
+            player:tradeComplete()
+            player:addKeyItem(ki)
+            player:setCharVar('LastMoogleKey', nextMidnight)
+            player:setCharVar('ASA_kit', 0)
```

Comment:
I don't know the current state of ASA in LSB, but this will eventually be replaced by the numbered vars of the interaction framework, so I'd look into what those will look like for their eventual migration

Reply:
I figured this would just be a placeholder until that happens (it'll likely be me and flibe doing them)